components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -1686,18 +1686,24 @@ public static NativeArray jsFunction_getAPI(Context cx, Scriptable thisObj,
                         row.put("userRate", row, userRate);
                     }
                     APIManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
-                    List<Environment> environments = config.getApiGatewayEnvironments();
-                    String envDetails = "";
-                    for (int i = 0; i < environments.size(); i++) {
-                        Environment environment = environments.get(i);
-                        envDetails += environment.getName() + ",";
-                        envDetails += filterUrls(environment.getApiGatewayEndpoint(), api.getTransports());
-                        if (i < environments.size() - 1) {
-                            envDetails += "|";
-                        }
+                    Map<String, Environment> environments = config.getApiGatewayEnvironments();
+                    StringBuilder envDetails = new StringBuilder();
+                    Set<String> environmentsPublishedByAPI =
+                            new HashSet<String>(api.getEnvironments());
+                    environmentsPublishedByAPI.remove("none");
+                    for (String environmentName : environmentsPublishedByAPI) {
+                        Environment environment = environments.get(environmentName);
+                        envDetails.append(environment.getName() + ",");
+                        envDetails.append(filterUrls(environment.getApiGatewayEndpoint(),
+                                                     api.getTransports()) + "|");
+                    }
+                    if (!"".equals(envDetails.toString())) {
```

Comment:
Can you please use isEmpty() method to check empty string, rather than using equals with ""?